src/cpp/common/channel_filter.cc

```diff
@@ -30,7 +30,7 @@ namespace grpc {
 grpc_linked_mdelem* MetadataBatch::AddMetadata(const string& key,
                                                const string& value) {
   grpc_linked_mdelem* storage = new grpc_linked_mdelem;
-  memset(storage, 0, sizeof(grpc_linked_mdelem));
+  memset(static_cast<void*>(storage), 0, sizeof(grpc_linked_mdelem));
```

Comment:
thank you for doing this. this memset is not needed because `grpc_linked_mdelem` has proper ctor now. can you please remove this and try msan tests?

`md` is set below, `next`/`prev` are set to `nullptr`, and reserved doesn't have to be set here.

Reply:
Thanks for checking in on us, now get back to vacation. I have a separate CL checking this out internally and I can try a PR for this purpose also.